test/gateway/eip20_cogateway/progress_redeem_with_proof.js

```diff
@@ -521,6 +531,69 @@ contract('EIP20CoGateway.progressRedeemWithProof() ', function (accounts) {
 
   });
 
+  it('should return penalty to redeemer when the message status in source is ' +
+    'declared revocation and in the target is progressed', async function () {
+    const facilitator = accounts[8];
+    const redeemer = redeemParams.redeemer;
+
+    await web3.eth.sendTransaction(
+      {
+        to: eip20CoGateway.address,
+        from: facilitator,
+        value: penaltyAmount,
+      }
+    );
+
+    await eip20CoGateway.setOutboxStatus(
+      redeemParams.messageHash,
+      MessageStatusEnum.DeclaredRevocation,
+    );
+
+    const initialFacilitatorEthBalance = await Utils.getBalance(facilitator);
+    const initialRedeemerEthBalance = await Utils.getBalance(redeemer);
+    const initialCoGatewayEthBalance = await Utils.getBalance(eip20CoGateway.address);
+
+    await setStorageRoot();
+
+    const tx = await eip20CoGateway.progressRedeemWithProof(
+      redeemParams.messageHash,
+      proofData.gateway.progress_unstake.proof_data.storageProof[0].serializedProof,
+      new BN(proofData.gateway.progress_unstake.proof_data.block_number, 16),
+      MessageStatusEnum.Progressed,
+      { from: facilitator },
+    );
+
+    const finalFacilitatorEthBalance = await Utils.getBalance(facilitator);
+    const finalRedeemerEthBalance = await Utils.getBalance(redeemer);
+    const finalCoGatewayEthBalance = await Utils.getBalance(eip20CoGateway.address);
+
+    const expectedFinalFacilitatorETHBalance = initialFacilitatorEthBalance
+      .add(bountyAmount)
+      .subn(tx.receipt.gasUsed);
```

Comment:
The gas price is set to `1` in `truffle.js`, which I think is the reason that it works. But yeah, in theory it would need to be multiplied with the gas price.